### PR TITLE
Remove linking and turn on hostname updating option

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request makes two changes:
* It removes the configuration of the data necessary to link the Nessus Agent and CDM Nessus server.
* It turns on a Nessus Agent option to send updates to the CDM Nessus server when the hostname changes.

## 💭 Motivation and context ##

We must remove the configuration of the Nessus Agent link from this Ansible role.  The linking process generates a Tenable Tag, which uniquely identifies the instance to the CDM Nessus server.  If we perform this step here, at AMI build time, then all instances created from that AMI will have identical Tenable Tags.  Nessus cannot handle "duplicated agents" like this, so we need to instead perform the linking via `cloud-init` at first boot.

Configuring the Nessus agent to send updates to the CDM Nessus server when the hostname changes is helpful because our instance's hostnames change when we join them to the FreeIPA domain.  This should make our instances show up in the CDM Nessus as `ipa0`, for example, instead of by the AWS-given hostname that is of the form `ip-10-20-30-40`.

See also:
* cisagov/freeipa-server-tf-module#45
* cisagov/freeipa-server-packer#51
* cisagov/cool-sharedservices-freeipa#36

## 🧪 Testing ##

All `pre-commit` hooks and `molecule` tests pass.  I also successfully built and deployed a new FreeIPA AMI for our COOL staging environment that includes these changes and verified that
* The FreeIPA instance launched from the new AMI continued to function as before
* The Nessus Agent is properly activated when said instance is spun up

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: molecule/default/tests/test_default.py [hunk 1]
```diff
@@ -22,22 +22,20 @@ def test_nessus_agent_installed(host):
...
-        ("ms_server_port", "8080"),
+        ("update_hostname", "yes"),
     ],
 )
 def test_nessus_config(host, key, value):
-    """Test that Nessus Agent is configured."""
+    """Test that Nessus Agent non-secure configuration options are as expected."""
     assert value in host.check_output(
-        f"/opt/nessus_agent/sbin/nessuscli fix --secure --get {key}"
+        f"/opt/nessus_agent/sbin/nessuscli fix --get {key}"
```
Comment (dav3r) [inline molecule/default/tests/test_default.py:40]: Can you let us know why the `--secure` flag was removed here?

**Review by dav3r — APPROVED** [review]
Looks good to me - I had one question, but it doesn't affect my approval here. 👍

File: molecule/default/tests/test_default.py (hunk 1 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py:40]: [The `--secure` flag](https://docs.tenable.com/nessusagent/Content/SettingsAdvanced.htm#SecureSettings) is used to interact with a subset of configuration options that `nessuscli` regards as sensitive.  It is not used for interacting with standard configuration options.

When I was performing the linking configuration in this role (i.e. in the code as it was before this PR) I was checking in the `molecule` tests that a list of four _secure_ (i.e. sensitive) configuration options that were part of the linking command were set as expected:  `groups`, `key`, `ms_server_ip`, and `ms_server_port`.  In this pull request I no longer set those four _secure_ options, but I added one _standard_ AKA _non-secure_ option:  `update_hostname`.  Now in the `molecule` tests I check that that one standard (i.e. not sensitive) option was set as expected; hence, I no longer need (or want) the `--secure` flag.

**Review by mcdonnnj — APPROVED** [review]
Thank you for your continued work on this... _situation_.

**Pull request merged**